### PR TITLE
packetbeat: wire default interface selection into config

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This allows users to specify sniffing on the default route interface for all platforms rather than having to specify a hard-coded interface or interface index on non-linux platforms.

## Why is it important?

This makes setting up large installations more robust for non-linux hosts as the interface identities do not need to be known during configuration.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Depends #31914 
- Updates #31905

### CODE REVIEW

File: packetbeat/docs/packetbeat-options.asciidoc [hunk 1]
```diff
@@ -122,6 +122,10 @@ packetbeat.interfaces.device: 0
 
 Specifying the index is especially useful on Windows where device names can be long.
 
+Alternatively the `default_route` device may be specified. This will set the capture
+device to be the device that is associated with the first default IPv4 route identified
+on Packetbeat start-up.
```
Comment (andrewkroh) [inline packetbeat/docs/packetbeat-options.asciidoc:127]: I'm thinking we should explicitly state that it won't change the monitored device after start-up.

**Review by andrewkroh — COMMENTED** [review]
LGTM

Do you want to add the default_route_ipv4 and default_route_ipv6 options here or will that be separate? (regarding https://github.com/elastic/beats/issues/31905#issuecomment-1154725592)

**Comment by efd6** [comment]
> Do you want to add the default_route_ipv4 and default_route_ipv6 options here or will that be separate?

This brings additional complexity that I would like to think about, so I was planning to do that in a later change.

File: packetbeat/docs/packetbeat-options.asciidoc (hunk 1 above)
Comment (efd6) [reply packetbeat/docs/packetbeat-options.asciidoc]: Added. This was my intention with the current wording, but I guess it is a little subtle.

**Review by andrewkroh — APPROVED** [review]
LGTM

**Pull request merged**